### DISCUSSION TITLE
Recent updates to "nan" package cause zlib-sync/erlpack to no longer be built properly

### DISCUSSION THREAD

**Discussion by JMTK** [discordjs/discord.js#8987] [General]
While not directly a discord.js problem, I thought it might be helpful to others who notice higher CPU usage when running their bots after regenerating their lock files recently. The postbuild script to rebuild those packages when installing discord.js (on Node 19 for me)
with:

```
../../nan/nan_callbacks.h:55:23: error: ‘AccessorSignature’ is not a member of ‘v8’
   55 | typedef v8::Local<v8::AccessorSignature> Sig;
      |                       ^~~~~~~~~~~~~~~~~
../../nan/nan_callbacks.h:55:40: error: template argument 1 is invalid
   55 | typedef v8::Local<v8::AccessorSignature> Sig;
      | 
```

Changelog: https://github.com/nodejs/nan/compare/v2.16.0...v2.17.0

`yarn why nan` showed that those packages prefer the older versions, simply adding to my package.json
```json
"resolutions": {
    "nan": "2.17.0"
},
```

fixes the issue. Not sure if there's anything better I could do

**Comment by vladfrangu** [comment]
Best bring this up to the respective packages (https://github.com/discord/erlpack and https://github.com/abalabahaha/zlib-sync respectively), as we cannot do anything about this ourselves (outside of forking the modules and building them ourselves)

**Comment by JMTK** [comment]
PRs created for reference:

- https://github.com/abalabahaha/zlib-sync/pull/9

- https://github.com/discord/erlpack/pull/58
